collections/sum_of.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+import { Selector } from "./types.ts";
+
+/**
+ * Applies the given selector to all elements in the given collection and calculates the sum of the results
+ *
+ * Example:
+ *
+ * ```typescript
```

Comment:
Apparently we have to use the `ts` tag instead of `typescript` for `deno test` to pick it up - and the rest of `std` uses `ts` as well. That was a mistake on my part in the original proposal, sorry.

Reply:
I'll change to `ts` for consistency but im pretty sure it picked up on it, as I got ci errors from that

Reply:
@kt3k mentioned it in my `mapNotNullish` PR, but it might already have been fixed